wire-ios/Wire-iOS UnitTests/ChangeEmailViewModelTests.swift

```diff
@@ -0,0 +1,118 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireSyncEngineSupport
+import XCTest
+
+@testable import Wire
+
+final class ChangeEmailViewModelTests: XCTestCase {
+
+    // MARK: - Properties
+
+    private var sut: ChangeEmailViewModel!
+    private var mockUserProfile: MockUserProfile!
+
+    // MARK: - setUp
+
+    override func setUp() {
+        super.setUp()
+        mockUserProfile = MockUserProfile()
+        sut = ChangeEmailViewModel(currentEmail: "current@example.com", userProfile: mockUserProfile)
+    }
+
+    // MARK: - tearDown
+
+    override func tearDown() {
+        sut = nil
+        mockUserProfile = nil
+        super.tearDown()
+    }
+
+    // MARK: - Unit Tests
+
+    func testRequestEmailUpdateWithInvalidEmail() {
+        // GIVEN
+        let expectation = self.expectation(description: "Email update completion")
```

Comment:
I _think_ we don't need to use expectations and waits in any of these tests as  `requestEmailUpdate` completes synchronously on the same thread. 

If you decide to keep using a completion handler in `requestEmailUpdate` (see my other comments in the view model), I would would create a helper method in this testsuite to convert the completion handler to a synchronous method  (`requestEmailUpdate() throws`). That way these tests will become much more concise.

Reply:
Same here: https://github.com/wireapp/wire-ios/commit/529ccb786e17c4b45e695aedb61feef240a9d6e8